pex/third_party/__init__.py

```diff
@@ -116,10 +120,11 @@ def iter_root_packages(self, relpath):
       yield package
 
   def _filter_names(self, relpath, pattern, group):
-    pat = re.compile(r'^{prefix}{pattern}$'
-                     .format(prefix=self.prefix + ((relpath + os.sep) if relpath else ''),
-                             pattern=pattern))
-
+    # We use '/' here instead of os.sep because the zip file format spec specifies that paths must
+    # use forward slashes. See section 4.4.17 of
+    # https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT
+    relpath_pat = '{}/'.format(relpath.replace(os.sep, '/')) if relpath else ''
```

Comment:
This might be an improvement. Inverts the expression so that there's an early return and applies `replace()` to the final result, rather than just to `relpath`.

```python
relpath_pat = '' if not relpath else '{}{}'.format(relpath, os.sep).replace(os.sep, '/')
```